### ISSUE TITLE
Error handling when applying incorrectly

### ISSUE THREAD

**Issue by nedbat** [boxed/mutmut#100]
After applying a mutation, I tried applying it again, and got an error:
```
$ mutmut apply 464
$ mutmut apply 464
Traceback (most recent call last):
  File "/usr/local/virtualenvs/cog/bin/mutmut", line 5, in <module>
    climain()
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 66, in wrapper
    f(*args, **kwargs)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 204, in climain
    version, suspicious_policy, untested_policy))
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 250, in main
    do_apply(argument, dict_synonyms, backup)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 123, in do_apply
    raise RuntimeError('No mutations performed. Are you sure the index is not too big?')
RuntimeError: No mutations performed. Are you sure the index is not too big?
```
I guess this is because the source line was not what the mutation expected?
But then I tried using an index that was too big as the error message suggested, and got a message not about an index too big:
```
$ mutmut apply 46444
Traceback (most recent call last):
  File "/usr/local/virtualenvs/cog/bin/mutmut", line 5, in <module>
    climain()
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 66, in wrapper
    f(*args, **kwargs)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 204, in climain
    version, suspicious_policy, untested_policy))
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 250, in main
    do_apply(argument, dict_synonyms, backup)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/__main__.py", line 112, in do_apply
    filename, mutation_id = filename_and_mutation_id_from_pk(int(mutation_pk))
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/cache.py", line 103, in wrapper
    return f(*args, **kwargs)
  File "<string>", line 2, in filename_and_mutation_id_from_pk
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/pony/orm/core.py", line 528, in new_func
    result = func(*args, **kwargs)
  File "/usr/local/virtualenvs/cog/lib/python3.7/site-packages/mutmut/cache.py", line 319, in filename_and_mutation_id_from_pk
    return mutant.line.sourcefile.filename, mutation_id_from_pk(pk)
AttributeError: 'NoneType' object has no attribute 'line'
```

**Comment by boxed** [comment]
Oops. Yea that error message is from an earlier version of mutmut where that made sense. Now the USA are just primary keys into an sqlite dB so doesn't make sense.

**Comment by boxed** [comment]
Fixed in b355a0c636d93704ffb6ade010595a8fa19c3213
